cinema/serializers.py

```diff
@@ -68,13 +80,65 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
 
+class TakenSeatsSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("row", "seat")
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TakenSeatsSerializer(
+        many=True,
+        read_only=True,
+        source="tickets"
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+        constraints = [UniqueConstraint(
+            fields=["movie_session", "row", "seat"],
+            name="unique_ticket_seat_trip")
+        ]
+
+
+class TicketListSerializer(TicketSerializer):
+    movie_session = MovieSessionListSerializer(
+        read_only=True,
+        many=False
+    )
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=False, allow_empty=False)
```

Comment:
The `TicketSerializer` is used with `read_only=False`, which implies that the incoming ticket data should be validated. However, the current implementation in the `create` method does not validate the ticket data before creating `Ticket` instances. Please ensure that the `tickets_data` is validated using `TicketSerializer` before creating `Ticket` objects.